src/prepare_pcap.c

```diff
@@ -349,53 +355,227 @@ struct dtmfpacket {
     struct rtpevent dtmf;
 };
 
-static u_long dtmf_ssrcid = 0x01020304;
+struct rtpnoop {
+    unsigned int request_rtcp:1;
+    unsigned int reserved:31;
+};
 
-void fill_default_dtmf(struct dtmfpacket* dtmfpacket, int marker, int seqno, int ts,
-                       char digit, int eoe, int duration)
-{
-    u_long pktlen = sizeof(struct dtmfpacket);
+struct nooppacket {
+    struct udphdr udp;
+    struct rtphdr rtp;
+    struct rtpnoop noop;
+};
+
+static u_long dtmf_ssrcid = 0x01020304; /* bug, should be random/unique */
 
+static void fill_default_udphdr(struct udphdr* udp, u_long pktlen)
+{
 #if defined(__HPUX) || defined(__DARWIN) || (defined __CYGWIN) || defined(__FreeBSD__)
-    dtmfpacket->udp.uh_ulen = htons(pktlen);
-    dtmfpacket->udp.uh_sum = 0 ;
-    dtmfpacket->udp.uh_sport = 0;
-    dtmfpacket->udp.uh_dport = 0;
+    udp->uh_ulen = htons(pktlen);
+    udp->uh_sum = 0 ;
+    udp->uh_sport = 0;
+    udp->uh_dport = 0;
 #else
-    dtmfpacket->udp.len = htons(pktlen);
-    dtmfpacket->udp.check = 0;
-    dtmfpacket->udp.source = 0;
-    dtmfpacket->udp.dest = 0;
+    udp->len = htons(pktlen);
+    udp->check = 0;
+    udp->source = 0;
+    udp->dest = 0;
```

Comment:
This is another thing (out of scope) if we compile with `DEFAULT_SOURCE`, you get BSD style definitions. We don't need this if/else.